tests/pytests/unit/renderers/test_stateconf.py

```diff
@@ -0,0 +1,364 @@
+import io
+import os
+import os.path
+import tempfile
+
+import pytest
+import salt.config
+import salt.loader
+from salt.exceptions import SaltRenderError
+from tests.support.runtests import RUNTIME_VARS
+
+REQUISITES = ["require", "require_in", "use", "use_in", "watch", "watch_in"]
+
+
+def _render_sls(content, sls="", saltenv="base", argline="-G yaml . jinja", **kws):
+    root_dir = tempfile.mkdtemp(dir=RUNTIME_VARS.TMP)
+    state_tree_dir = os.path.join(root_dir, "state_tree")
+    cache_dir = os.path.join(root_dir, "cachedir")
+    if not os.path.isdir(root_dir):
+        os.makedirs(root_dir)
+
+    if not os.path.isdir(state_tree_dir):
+        os.makedirs(state_tree_dir)
+
+    if not os.path.isdir(cache_dir):
+        os.makedirs(cache_dir)
+    config = salt.config.minion_config(None)
+    config["root_dir"] = root_dir
+    config["state_events"] = False
+    config["id"] = "match"
+    config["file_client"] = "local"
+    config["file_roots"] = dict(base=[state_tree_dir])
+    config["cachedir"] = cache_dir
+    config["test"] = False
+    _renderers = salt.loader.render(config, {"config.get": lambda a, b: False})
+
+    return _renderers["stateconf"](
+        io.StringIO(content),
+        saltenv=saltenv,
+        sls=sls,
+        argline=argline,
+        renderers=salt.loader.render(config, {}),
+        **kws
+    )
+
+
+def test_state_config():
+    result = _render_sls(
+        """
```

Comment:
```suggestion
def test_state_config(renderer):
    result = renderer(
        """
```
See where I'm getting at?